relayer/src/foreign_client.rs

```diff
@@ -213,16 +213,16 @@ define_error! {
 }
 
 #[derive(Clone, Debug)]
-pub struct ForeignClient {
+pub struct ForeignClient<Host: ChainHandle, Guest: ChainHandle> {
```

Comment:
I like the idea of trying to be more principled about terminology.

I think this PR could be a good place to gather input about all the places in the code where we have exceptions and inconsistencies in our terminology. It's also a good opportunity to see all the different naming scheme we have for the same concept:
- source + destination
- host + target
- party + counterparty
- chainA + chainB
- host + guest


I think a good naming scheme (1) is simple & intuitive, and (2) we can apply it without exceptions (we can cover as many of the concepts above as possible). I wouldn't be too ambitious about introducing a new scheme, or trying to settle on any just yet.